src/atomic/SDL_spinlock.c

```diff
@@ -160,8 +142,22 @@ SDL_bool SDL_AtomicTryLock(SDL_SpinLock *lock)
     }
     return res;
 #else
-#error Please implement for your platform.
-    return SDL_FALSE;
+    /* Terrible terrible damage */
+    static SDL_Mutex *_spinlock_mutex;
+
+    if (!_spinlock_mutex) {
+        /* Race condition on first lock... */
+        _spinlock_mutex = SDL_CreateMutex();
+    }
+    SDL_LockMutex(_spinlock_mutex);
+    if (*lock == 0) {
+        *lock = 1;
+        SDL_UnlockMutex(_spinlock_mutex);
+        return SDL_TRUE;
+    } else {
+        SDL_UnlockMutex(_spinlock_mutex);
+        return SDL_FALSE;
+    }
```

Comment:
Is this really safe? In the case where we're loading a newer SDL via DYNAPI (let's say we are SDL 3.2 and we're loading SDL 3.42 via DYNAPI), can't we have this situation?

- `SDL_CreateMutex()` creates a v3.2 mutex, locks it and unlocks it
- we initialize DYNAPI and release `*lock`
- next time we get here, `SDL_LockMutex()` is the v3.42 version but we are asking it to lock a v3.2 mutex, which has a different in-memory layout
- kaboom?

Reply:
For internal SDL code, all public functions are defined as XXX_REAL so they won't resolve to the public symbols, so I think this is safe.